pkg/image/controller/signature/container_image_downloader.go

```diff
@@ -30,9 +32,19 @@ func (s *containerImageSignatureDownloader) DownloadImageSignatures(image *image
 	if err != nil {
 		return nil, err
 	}
-	source, err := reference.NewImageSource(nil, nil)
+	// TODO: We just need to trick ping() to pass and not error, we don't care
+	// whether the registry is secure or not at this point as GetSignatures() use
+	// different client for signature retrieval?
+	sctx := types.SystemContext{
+		DockerInsecureSkipTLSVerify: true,
+	}
+	source, err := reference.NewImageSource(&sctx, nil)
 	if err != nil {
-		return nil, err
+		// In case we fail to talk to registry to get the image metadata (private
+		// registry, internal registry, etc...), do not fail with error to avoid
+		// spamming logs.
+		glog.V(4).Infof("Failed to get %q: %v", image.DockerImageReferenc, err)
```

Comment:
@miminar @mtrmac @bparees  i see billion of errors in online where we trying to contact the internal registry (?) in `NewImageSource` and failing with error (which causing retries). I think we can safely ignore that error for now because we only care about RH signatures at this point. Also I don't understand why we need to do ping() for the registry when we want only get the signatures (which are downloaded from other source anyway).

example of error:
```
Oct 19 00:38:48 XXXX.ca-central-1.compute.internal atomic-openshift-master-controllers[4314]: E1019 00:38:48.538149    4314 signature_import_controller.go:109] error syncing image sha256:XXXXX, it will be retried: unauthorized: authentication required
```

@mtrmac can we make some option in system context that will skip all remote communication to registry?

Reply:
@smarterclayton found why signature import was pretty high in heap :)

Reply:
> @mtrmac can we make some option in system context that will skip all remote communication to registry?

Not really; the client needs to connect to `/v2` to see whether the server supports `X-Registry-Supports-Signaturs`, and based on that decide which signature access client to use. 

In https://github.com/openshift/origin/pull/15494#discussion_r130623744 I was explicit that this is relying on an implementation detail, and that the controller should instead supply the appropriate credentials. That’s eventually going to have to happen anyway, AFAICT, so I’d really hate to extend this hack any further just to postpone doing the right thing.